ql/experimental/asian/analytic_discr_geom_av_price_heston.cpp

```diff
@@ -0,0 +1,285 @@
+/* -*- mode: c++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*
+ Copyright (C) 2020 Jack Gillett
+ 
+ This file is part of QuantLib, a free-software/open-source library
+ for financial quantitative analysts and developers - http://quantlib.org/
+
+ QuantLib is free software: you can redistribute it and/or modify it
+ under the terms of the QuantLib license.  You should have received a
+ copy of the license along with this program; if not, please email
+ <quantlib-dev@lists.sf.net>. The license is also available online at
+ <http://quantlib.org/license.shtml>.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ FOR A PARTICULAR PURPOSE.  See the license for more details.
+*/
+
+#include <ql/experimental/asian/analytic_discr_geom_av_price_heston.hpp>
+
+namespace QuantLib {
+
+    // A class to perform the integrations in Eqs (23) and (24)
+    class AdgapIntegrand {
```

Comment:
Since this class is meant to only be used internally by this file, it can be declared inside an anonymous namespace to avoid future name clashes during linking.

Reply:
Thanks, fixed this for this pricer and also for the pricer it was clashing with

Reply:
This has caused some issues with the Unity build (clashing definitions of `Integrand`), and I see this is a known problem - eg. https://engineering.leewinder.co.uk/2009/12/15/the-evils-of-unity-builds/. How should I approach this? I'll need to import both of these classes into the `test-suite` so I don't think I can fully isolate them unless we break that into two?

The two classes calculate a similar integrand so I could move them into a shared utility class. They both call a method of the parent class which needs slightly different parameters in each case (eg. fixing times in the discrete case), but I could set these as mutable class properties inside the `calculate()` method once the instrument arguments have been provided to the pricing engine.

It feels like this might be a bit over-the-top though?

Reply:
One simple way to fix it would be to use two different names for the inner namespace, instead of using anonymous ones; another would be to give the classes two different names.